erpnext/hr/doctype/leave_policy_assignment/leave_policy_assignment.py

```diff
@@ -10,31 +10,39 @@
 from math import ceil
 import json
 from six import string_types
+from frappe.utils import comma_and
 
 class LeavePolicyAssignment(Document):
 
 	def validate(self):
 		self.validate_policy_assignment_overlap()
 		self.set_dates()
 
+	def on_submit(self):
+		self.grant_leave_alloc_for_employee()
+
+	def get_allocation(self):
+		return frappe.get_all("Leave Allocation", filters = {"leave_policy_assignment": self.name})
+
 	def set_dates(self):
 		if self.assignment_based_on == "Leave Period":
 			self.effective_from, self.effective_to = frappe.db.get_value("Leave Period", self.leave_period, ["from_date", "to_date"])
 		elif self.assignment_based_on == "Joining Date":
 			self.effective_from = frappe.db.get_value("Employee", self.employee, "date_of_joining")
 
 	def validate_policy_assignment_overlap(self):
-		leave_policy_assignments = frappe.get_all("Leave Policy Assignment", filters = {
-			"employee": self.employee,
-			"name": ("!=", self.name),
-			"docstatus": 1,
-			"effective_to": (">=", self.effective_from),
-			"effective_from": ("<=", self.effective_to)
-		})
-
-		if len(leave_policy_assignments):
-			frappe.throw(_("Leave Policy: {0} already assigned for Employee {1} for period {2} to {3}")
-				.format(bold(self.leave_policy), bold(self.employee), bold(formatdate(self.effective_from)), bold(formatdate(self.effective_to))))
+		if self.docstatus == 1:
+			leave_policy_assignments = frappe.get_all("Leave Policy Assignment", filters = {
+				"employee": self.employee,
+				"name": ("!=", self.name),
+				"docstatus": 1,
+				"effective_to": (">=", self.effective_from),
+				"effective_from": ("<=", self.effective_to)
+			})
+
+			if len(leave_policy_assignments):
+				frappe.throw(_("Leave Policy: {0} already assigned for Employee {1} for period {2} to {3}")
+					.format(bold(self.leave_policy), bold(self.employee), bold(formatdate(self.effective_from)), bold(formatdate(self.effective_to))))
 
 	def grant_leave_alloc_for_employee(self):
```

Comment:
Steps:
1. Select employee.
2. Add date range.
3. change Assignment based on to joining.
4. Again change assignment based on to leave period.
5. select a leave period from the list
5. Again change assignment based on to blank.
6. When saving or submitting assignment based on leave period still exist because it is set in leave allocation

Reply:
fixed